File: drawfunctions.lua

module(..., package.seeall)

local squareWidth = 0
local squareHeight = 0 

function drawCirlce(x, y, xoffset, yoffset)
   local circle = display.newCircle(x + xoffset *squareWidth, y+ yoffset *squareHeight, squareWidth/3)
   circle:setFillColor(0,0,0,0)
   circle.strokeWidth = 7
   circle:setStrokeColor(255, 255,0)
   return circle
end

function drawCross(xMin, yMin, xMax, yMax, xoffset, yoffset)
    local notedge = 15
    
    print(math.ceil(xMin + xoffset *squareWidth + notedge))
    print(math.ceil(yMin + yoffset *squareHeight +notedge))
    print(math.ceil(xMax + xoffset *squareWidth -notedge))
    print(math.ceil(yMax + yoffset *squareHeight -notedge))
    



    local line = display.newLine(math.ceil(xMin + xoffset *squareWidth + notedge), math.ceil(yMin + yoffset *squareHeight +notedge), math.ceil(xMax + xoffset *squareWidth -notedge), math.ceil(yMax + yoffset *squareHeight -notedge))
    local line2 = display.newLine(xMin + xoffset *squareWidth + notedge, yMax + yoffset *squareHeight -notedge, xMax + xoffset *squareWidth - notedge, yMin + yoffset *squareHeight + notedge)


--			     local line = display.newLine(rec.contentBounds.xMin + xposoffset *squareWidth, rec.contentBounds.yMin + yposoffset *squareHeight, rec.contentBounds.xMax + xposoffset *squareWidth, rec.contentBounds.yMax + yposoffset *squareHeight)
--	             local line2 = display.newLine(rec.contentBounds.xMin + xposoffset *squareWidth, rec.contentBounds.yMax + yposoffset *squareHeight, rec.contentBounds.xMax + xposoffset *squareWidth, rec.contentBounds.yMin + yposoffset *squareHeight)
	line.width = 7
	line2.width = 7
	line:setColor(0,255,255)
	line2:setColor(0,255,255)
	
    local crossgroup  = display.newGroup()
    crossgroup:insert(line)
    crossgroup:insert(line2)
    
    return crossgroup

end

function setValues(params)
    squareWidth = params.squareWidth
    squareHeight = params.squareHeight


end

File: drawfunctions.js
// drawfunctions.js - Drawing utilities for the game

class DrawFunctions {
    constructor(squareWidth, squareHeight) {
        this.squareWidth = squareWidth;
        this.squareHeight = squareHeight;
    }

    setValues(squareWidth, squareHeight) {
        this.squareWidth = squareWidth;
        this.squareHeight = squareHeight;
    }

    drawCircle(ctx, x, y, xOffset, yOffset) {
        const centerX = x + xOffset * this.squareWidth;
        const centerY = y + yOffset * this.squareHeight;
        const radius = this.squareWidth / 3;

        ctx.beginPath();
        ctx.arc(centerX, centerY, radius, 0, 2 * Math.PI);
        ctx.strokeStyle = '#ffff00'; // Yellow
        ctx.lineWidth = 7;
        ctx.stroke();
    }

    drawCross(ctx, xMin, yMin, xMax, yMax, xOffset, yOffset) {
        const notEdge = 15;
        
        const x1 = xMin + xOffset * this.squareWidth + notEdge;
        const y1 = yMin + yOffset * this.squareHeight + notEdge;
        const x2 = xMax + xOffset * this.squareWidth - notEdge;
        const y2 = yMax + yOffset * this.squareHeight - notEdge;

        ctx.strokeStyle = '#00ffff'; // Cyan
        ctx.lineWidth = 7;
        
        // Draw first line (top-left to bottom-right)
        ctx.beginPath();
        ctx.moveTo(x1, y1);
        ctx.lineTo(x2, y2);
        ctx.stroke();

        // Draw second line (bottom-left to top-right)
        ctx.beginPath();
        ctx.moveTo(x1, y2);
        ctx.lineTo(x2, y1);
        ctx.stroke();
    }
}
